### ISSUE TITLE
"You're using Python 0.0"

### ISSUE THREAD

**Issue by baylf2000** [microsoft/PTVS#6841]
**Describe the bug**
When using Python 3.10, Visual Studio gives the error message "You're using Python 0.0. Some features might not work as expected since Visual Studio no longer supports this version." on startup, as well as two Pylance popup errors.

**Steps to Reproduce**
1. Install Python 3.10, or use Python 3.10 in an environment.

**Additional context and screenshots**
![image](https://user-images.githubusercontent.com/1748322/148155007-4d9ad225-6ed0-46cd-910d-2b652e472899.png)

**Configuration information (If you are providing a diagnostics file (see below), skip this section)**
VS Version: `Microsoft Visual Studio Community 2022 (64-bit) - Version 17.0.4`
PTVS version: 
```
Python - Django support   17.0.21293.2
Provides templates and integration for the Django web framework.

Python - Profiling support   17.0.21293.2
Profiling support for Python projects.

Python - VC Project Support   17.0.21153.2
Provides support for launching C++ projects with Python debugging enabled.

Python with Pylance   17.0.21293.2
Provides IntelliSense, projects, templates, debugging, interactive windows, and other support for Python developers.```

Python version: 3.10

**Comment by ark2git** [comment]
+ 1
I’m experiencing the same issue

**Comment by baylf2000** [comment]
Seems the exact same issue happened when Python 3.9 was released.

**Comment by AdamYoblick** [comment]
Dupe of #6795

**Comment by kolkinath** [comment]
""You're using Python 0.0. Some features might not work as expected since Visual Studio no longer supports this version. More info Don't show this message again"" ---WHAT IS WRONG WITH THIS!?

**Comment by AdamYoblick** [comment]
Hi @kolkinath , it's a bug. Python versions are listed in a enum, and the default maj/min versions are 0, which is the cause of the strange error message. I agree, the message should be much better.

I'm actually in the process of fixing this bug and adding support for python 3.10. It's a bit more work than just adding to an enum, since we have proxy classes that map 1:1 to python c structs, and the member offsets need to be the same for profiling to work. I will reply back once 3.10 support is added, but it will take a little time for the change to make its way through to a VS release. Cheers 😄

**Comment by Tayinator** [comment]
How do we fix it in the meantime?

**Comment by AdamYoblick** [comment]
Hi @Tayinator , there is no workaround because of the reasons I mentioned above. It’s more than just a simple enum addition. Please use 3.9 or lower until the fix is in. 

Our test team also found a bug in anaconda environments using python 3.10 which is actually a conda bug, but we’re trying a workaround to see how it goes.

**Comment by baylf2000** [comment]
@AdamYoblick 

I don't seek to be critical, but I have to ask the question... how long will it take to fix this? 

- Python 3.10 was released over six months ago.
- Visual Studio is the premier development tool for the largest software company in the world.
- Python is widely accepted as the most popular programming language in the world.

Am I correct in saying that your team is the official team employed by Microsoft for this project?

How difficult can it be to add support for a simple version release? Will it take another six months or more for version 3.11?

Perhaps if the addition of version support to PVTS can't even keep up with the official Python release schedule then the whole project should be re-examined? Other IDE's support new Python releases either immediately or within a week or two. Why is VS any different?

**Comment by AdamYoblick** [comment]
This issue is closed, meaning 3.10 support has been added and will be in the next release of Visual Studio. 😄 
There are some bugs in Anaconda environments due to bugs in conda itself, and we are looking at implementing a workaround in the meantime. Hopefully that workaround will make it into the same release.

Regarding your other questions:

- Yes, our team is the Microsoft PTVS team. But the team also supports debugpy (python debugger) and pylance (the python language server), so we have many things on our plate and only so much capacity.
- The difficulty of adding support for a version release depends on what changes are made in that release. As I mentioned previously, it's not just a simple enum addition. We have classes that wrap python c structs and the offsets need to be identical or profiling will not work. From what I've heard, the changes from 3.10 -> 3.11 are much more significant than 3.9 -> 3.10. I believe the stack frame implementation has been overhauled (amongst other things), which means our debugger and profiler need to be changed to match.
- I apologize if you feel like the project should be re-examined. There are things like team capacity and priorities that sometimes affect how quickly we can add new features to the product.

**Comment by John1900** [comment]
I just installed VC 2022 and get the same error  "Your're using Python 0.0.....".  Help/Check for updates shows version is current 17.1.3   2022 4 3

Thanks

**Comment by TylerPantuso** [comment]
Still an issue

**Comment by AdamYoblick** [comment]
@TylerPantuso Have you tried the latest *preview* version of VS? It's fixed there, or you can wait for the 17.2 release. Thanks!

**Comment by delizin** [comment]
I am still getting this bug using VS Community 2022 17.2.5 with a Python 3.10 environment on an existing project.

**Comment by AdamYoblick** [comment]
@delizin Have you tried the latest preview version of VS? If yes, can you please provide a sample project where this is happening to you, as we can no longer repro it.

**Comment by delizin** [comment]
@AdamYoblick Going forward this issue seems to be fixed in 17.2.5. In my existing project's .pyproj file, the interpreter version was improperly set to 0.0 and it never updated. 

```
    <Interpreter Include="env\">
      <Id>env</Id>
      <Version>0.0</Version>
      <Description>env (Python 3.10 (64-bit))</Description>
      <InterpreterPath>Scripts\python.exe</InterpreterPath>
      <WindowsInterpreterPath>Scripts\pythonw.exe</WindowsInterpreterPath>
      <PathEnvironmentVariable>PYTHONPATH</PathEnvironmentVariable>
      <Architecture>X64</Architecture>
    </Interpreter>
```

Manually changing the version to 3.10 fixes the problem for existing projects.
